modules/disk-uploader/pkg/vmexport/vmexport.go

```diff
@@ -99,6 +99,43 @@ func GetRawDiskUrlFromVolumes(client kubecli.KubevirtClient, namespace, name, vo
 	return "", fmt.Errorf("volume %s is not found in VirtualMachineExport internal volumes", volumeName)
 }
 
+func GetLabelsFromExportSource(k8sClient *k8sv1.CoreV1Client, virtClient kubecli.KubevirtClient, exportSourceKind, exportSourceNamespace, exportSourceName, volumeName string) (map[string]string, error) {
+	source, err := getExportSource(exportSourceKind, exportSourceName)
+	if err != nil {
+		return nil, err
+	}
+
+	var sourceLabels map[string]string
+	switch source.Kind {
```

Comment:
This duplicates the switch in `getExportSource`, I'd remove the call to `getExportSource` and just switch on `exportSourceKind` here.

Reply:
Yep!